components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java

```diff
@@ -83,7 +85,7 @@ protected void doStart() throws Exception {
         if (jobLauncher == null) {
             jobLauncher = resolveJobLauncher();
         }
-        if (job == null && jobName != null) {
+        if (job == null && jobName != null && jobName.compareTo("dynamic") != 0) {
```

Comment:
of course there's something wrong, it means that you need to know beforehand a existing job... look at the line of code blow the if, if the job is dynamic, then it would try to find a job (therefore not dynamic). Fixed the equals.